src/Features/Core/Portable/CodeRefactorings/SyncNamespace/AbstractChangeNamespaceService.cs

```diff
@@ -102,6 +106,67 @@ public override async Task<bool> CanChangeNamespaceAsync(Document document, Synt
             return !applicableContainers.IsDefault;
         }
 
+        public override string TryBuildNamespaceFromFolders(IEnumerable<string> folders, ISyntaxFacts syntaxFacts)
+        {
+            var parts = folders.SelectMany(folder => folder.Split(new[] { '.' })).SelectAsArray(EscapeIdentifier);
+            return parts.All(syntaxFacts.IsValidIdentifier) ? string.Join(".", parts) : null;
```

Comment:
isn't this logic the same for VB and C#?  not sure why this needs to be on the interface and can't just be pulled into the caller.

Reply:
This gets used in a few places and I wanted it to be the same everywhere. Is there a better place for a helper like this?

Reply:
It is. It's used a few places and I wanted it to be the same. Is there a better place for a helper style method?

Reply:
Move to PathUtilities.